### PR TITLE
Prevents contest stats page crash; Fixes #1229

### LINKED ISSUE #1229: Stats page crashes if contest has no problems
If a contest has no problems, the stats page will 500.

Stacktrace:
```
Traceback:
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
 34.             response = get_response(request)
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
 115.                 response = self.process_exception_by_middleware(e, request)
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
 113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/views/generic/base.py" in view
 71.             return self.dispatch(request, *args, **kwargs)
File "/code/site/judge/views/contests.py" in dispatch
 216.             return super(ContestMixin, self).dispatch(request, *args, **kwargs)
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/views/generic/base.py" in dispatch
 97.         return handler(request, *args, **kwargs)
File "/code/dmoj-virtenv/lib/python3.7/site-packages/django/views/generic/detail.py" in get
 107.         context = self.get_context_data(object=self.object)
File "/code/site/judge/views/contests.py" in get_context_data
 519.             *self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code'),
Exception Type: ValueError at /contest/musmid/stats
Exception Value: not enough values to unpack (expected 2, got 0)
```

### PR DESCRIPTION
**Pull request by Carson-Tang**

### CODE REVIEW

File: judge/views/contests.py [hunk 1]
```diff
@@ -510,9 +510,11 @@ def get_context_data(self, **kwargs):
             queryset.values('problem__code', 'result').annotate(count=Count('result'))
                     .values_list('problem__code', 'result', 'count'),
         )
-        labels, codes = zip(
-            *self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code'),
-        )
+        labels, codes = [], []
+        if len(self.object.contest_problems.order_by('order')) > 0:
```
Comment (kiritofeng) [inline judge/views/contests.py]: Please use an `exists()` query instead

Comment (quantum5) [inline judge/views/contests.py]: Highly inefficient, should just store `self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code')` into a variable.

File: judge/views/contests.py [hunk 2]
```diff
@@ -510,9 +510,12 @@ def get_context_data(self, **kwargs):
             queryset.values('problem__code', 'result').annotate(count=Count('result'))
                     .values_list('problem__code', 'result', 'count'),
         )
-        labels, codes = zip(
-            *self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code'),
-        )
+        labels, codes = [], []
+        contest_problems = self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code')
+        if contest_problems.exists():
```
Comment (quantum5) [inline judge/views/contests.py]: Why are you using `.exists()`? You are fetching it anyways so might as well hit the database once...

```suggestion
        if contest_problems:
```

File: judge/views/contests.py [hunk 3]
```diff
@@ -510,9 +510,12 @@ def get_context_data(self, **kwargs):
             queryset.values('problem__code', 'result').annotate(count=Count('result'))
                     .values_list('problem__code', 'result', 'count'),
         )
-        labels, codes = zip(
-            *self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code'),
-        )
+        labels, codes = [], []
+        contest_problems = self.object.contest_problems.order_by('order').values_list('problem__name', 'problem__code')
+        if contest_problems:
+            labels, codes = zip(
```
Comment (Xyene) [inline judge/views/contests.py]: This `zip` can be on one line now that it's not super long.

**Pull request merged**